### DISCUSSION TITLE
How are scores calculated?

### DISCUSSION THREAD

**Discussion by acarr** [Myndex/SAPC-APCA#46] [Q&A] [answered]
Thank you for all this work and I'm really excited about putting some of these ideas to use!

While doing that, I've been reading through the documentation and have not been able to grasp how the scores are calculated (5–0). I saw the old [look-up tables](https://github.com/Myndex/SAPC-APCA/discussions/27#discussioncomment-1565001) but I believe those are now obsolete. Any new guidance on how those should be calculated?

**Comment by Myndex** [comment, marked as answer]
Hi Alex @acarr  

For the time being, I suggest just looking at "Score 4" as the ideal minimum score to keep things simple. The multi level scoring is a function of WCAG 3.0, and still in development for the standard in general.

The [definitions are on the Wiki ](https://www.w3.org/WAI/GL/task-forces/silver/wiki/Visual_Contrast_of_Text_Subgroup#Draft_Conformance_Model)but have not been revisited since early in the year. 

The "most recent" model is described on the [APCA demo tool page](https://www.myndex.com/APCA/), again, subject to change.

What I have been working on more is the use-case examples. I think defining use cases is a better model moving forward. You can read more about that in discussion #39  https://github.com/Myndex/SAPC-APCA/discussions/39


Thank you!

Andy

**Reply by acarr** [reply]
Thanks Andy @Myndex!

> What I have been working on more is the use-case examples. I think defining use cases is a better model moving forward. You can read more about that in discussion #39

Do you think the best way to represent readability/conformance is to highlight the use case and the minimum contrast range? Similar to what we do with WCAG 2.0 in that we say "aim for this contrast for _large text_ and this contrast for _normal text_" we'd say "_Body Text_ should be min Lc 75 and _navigation text_ should be min Lc 60" etc... I'm thinking in terms of how to best convey this info to designers in a contrast testing tool. Score feels easier to comprehend but on face value it might not take into account their use-case.

> The "most recent" model is described on the APCA demo tool page, again, subject to change.

I guess my confusion is how the model on the APCA page with the descriptions for 90, 75, 60, 45, interacts with the lookup table. Those definitions seem like general guidelines but how do they relate to the lookup table where some of the minimums are as low as 45? Is the lookup table representative of a "score of 4" or is it something different all together?

Also, I missed the levels as listed on the SAPCA tool and was only looking at the APCA demo which highlights "Level 4" at the top. How are scores calculated in that tool? 

_Sorry for the 20 questions. Trying to make sure I understand these details. Appreciate all of this work!_

**Reply by Myndex** [reply]
Hi Alex!! @acarr Don't apologize for questions here, all are welcome!

**And so here's the thing:** contrast is not about color. And, color is not "real", neither is "contrast".

The human vision system (HVS) is massively complex, abstract, and unintuitive. 62% of the brain is somehow involved in visual processing, and over 22% is dedicated to vision. Most of what you "see" is really a product of your neurology, not so much what the eyes "receive". 

A first critical thing to know: contrast sensitivity (i.e. how "contrasty" something appears) is more closely related to spatial frequency — for our purposes this is the weight and size of a font. A higher spatial frequency (i.e. a smaller and thinner font) has a lower contrast, so the colors used need to be increased in contrast to compensate.

APCA contrast guidelines for readability are spatially aware. Lc45 is fine for a color contrast **_WHEN_** the object seen is large enough and thick enough. Lc45 is not at all useful though for readability for blocks of body text.

As for use cases, we have different needs in terms of our perception. For full speed fluent reading, we need a higher contrast than, say, for "spot" reading where we might be looking at just a word or two. We have a different level of contrast needed to discern edges of a thick button or form field. This is one of the ways WCAG_2 contrast fails in that it tries to fit "everything" into a couple contrast levels, serving none of them well.

As for the SAPC tool and the "many levels" this was an early demonstrator of a possible multi-level method, which was also intended to help existing sites that pass WCAG_2 continue to pass. It seems to complicated for some though, so I made the APCA tool to simplify it while maintaining the essence of variable font minimums.


I hope this answers your questions!

Thank you

Andy
